compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -600,8 +601,9 @@ class TypeErasure(sourceLanguage: SourceLanguage, semiEraseVCs: Boolean, isConst
       assert(refinedInfo.isInstanceOf[PolyType])
       val res = refinedInfo.resultType
       val paramss = res.paramNamess
-      assert(paramss.length == 1)
-      this(defn.FunctionType(paramss.head.length, isContextual = res.isImplicitMethod, isErased = res.isErasedMethod))
+      val arity = if paramss.isEmpty then 0 else paramss.head.length
+      assert(arity <= 1)
```

Comment:
That assertion looks like it would prevent a polymorphic function with more than one value parameter.

Reply:
Oops, you are right.